src/submission/validation-clinical/utils.ts

```diff
@@ -516,44 +516,31 @@ export function getRelatedEntityByFK(
 	return entity;
 }
 
-export const getDataFromDonorRecordOrDonor = (
+/**
+ * Get vitalStatus and survivalTime from a donor. Return undefined if either are missing.
+ */
+export const getSurvivalDataFromDonor = (
 	submittedRecord: DeepReadonly<SubmittedClinicalRecord>,
 	donor: DeepReadonly<Donor>,
-	errors: SubmissionValidationError[],
 	fieldName: ClinicalFields,
-) => {
-	let missingDonorFields: string[] = [];
+): undefined | { donorVitalStatus: string; donorSurvivalTime: number } => {
 	let donorVitalStatus: string = '';
 	let donorSurvivalTime: number = NaN;
 	const donorDataSource = donor.clinicalInfo || {};
 
 	if (_.isEmpty(donorDataSource)) {
-		missingDonorFields = [DonorFieldsEnum.vital_status, DonorFieldsEnum.survival_time];
+		return undefined;
 	} else {
 		donorVitalStatus = (donorDataSource[DonorFieldsEnum.vital_status] as string) || '';
 		donorSurvivalTime = Number(donorDataSource[DonorFieldsEnum.survival_time]) || NaN;
 
 		if (isEmptyString(donorVitalStatus)) {
-			missingDonorFields.push(DonorFieldsEnum.vital_status);
+			return undefined;
 		}
 		if (donorVitalStatus.toString().toLowerCase() === 'deceased' && isNaN(donorSurvivalTime)) {
-			missingDonorFields.push(DonorFieldsEnum.survival_time);
+			return undefined;
 		}
 	}
 
-	if (missingDonorFields.length > 0) {
-		errors.push(
-			buildSubmissionError(
-				submittedRecord,
-				DataValidationErrors.NOT_ENOUGH_INFO_TO_VALIDATE,
```

Comment:
Source of testing errors -- this is removed and not replaced elsewhere. The tests are looking for the `NOT_ENOUGH_INFO_TO_VALIDATE` error and it's not found.